src/containerapp/azext_containerapp/tests/latest/test_containerapp_patch.py

```diff
@@ -0,0 +1,204 @@
+        ingress = 'external'
+        target_port = '80'
+
+        # Generate a name for the Container App
+        app_name = self.create_random_name(prefix='containerapp', length=24)
+
+        # Create managed environment
+        env_name = self.create_random_name(prefix='env', length=24)
+        self.cmd(f'containerapp env create -g {resource_group} -n {env_name}')
+
+        self._create_and_verify_containerapp_up(resource_group,env_name=env_name,source_path=source_path,ingress=ingress,target_port=target_port,app_name=app_name)
+        self._retag_image_to_older_version_and_push(resource_group=resource_group,app_name=app_name)
+
+        # Execute and verify patch list command
+        patchable_images = self.cmd(f'containerapp patch list -g {resource_group} --environment {env_name}').get_output_in_json()
+        newRunImageTag = patchable_images[0]["newRunImage"].split(':')[1]
+        self.assertEquals(newRunImageTag,_utils.get_latest_buildpack_run_tag("aspnet", "7.0"))
+        self.assertEquals(patchable_images[0]["oldRunImage"], "run-dotnet-aspnet-7.0.1-cbl-mariner2.0")
+
+        # Execute and verify patch apply command
+        self.cmd(f'containerapp patch apply -g {resource_group} --environment {env_name}')
+        app = self.cmd(f"containerapp show -g {resource_group} -n {app_name}").get_output_in_json()
+        patched_image = app["properties"]["template"]["containers"][0]["image"]
+        patched_image_tag = patched_image.split(':')[1]
+        self.assertEquals(patched_image_tag,_utils.get_latest_buildpack_run_tag("aspnet", "7.0"))
+
+    @live_only()
+    @ResourceGroupPreparer(location = "eastus2")
+    def test_containerapp_patch_list_and_apply_with_show_all_e2e(self, resource_group):
+        image = "mcr.microsoft.com/k8se/quickstart:latest"
+
+        # Generate a name for the Container App
+        app_name = self.create_random_name(prefix='containerapp', length=24)
+
+        # Create managed environment
+        env_name = self.create_random_name(prefix='env', length=24)
+        self.cmd(f'containerapp env create -g {resource_group} -n {env_name}')
+
+        self._create_and_verify_containerapp_up(resource_group,env_name=env_name,image=image,app_name=app_name)
+
+        # Execute and verify patch list command
+        patch_cmd = f'containerapp patch list -g {resource_group} --environment {env_name} --show-all'
+        output = self.cmd(patch_cmd).get_output_in_json()
+        self.assertEquals(output[0]["targetImageName"],"mcr.microsoft.com/k8se/quickstart:latest")
+
+        # Execute and verify patch apply command
+        self.cmd(f'containerapp patch apply -g {resource_group} --environment {env_name} --show-all')
+        app = self.cmd(f"containerapp show -g {resource_group} -n {app_name}").get_output_in_json()
+        image = app["properties"]["template"]["containers"][0]["image"]
+        self.assertEquals(image,"mcr.microsoft.com/k8se/quickstart:latest")
+
```

Comment:
I'm wondering if we could deploy 2 container apps (one with quick start image, one with an old patchable image) to make sure that the `--show-all` is working properly by adding assert to check how many results have been returned by the `patcher list` command.

Reply:
@harryli0108 In my opinion, it may be better to explicitly check for the image name than the count of container apps returned from patch list. The ordering may not be the same every time if we have multiple apps and would make it difficult to assert against the image name. I would think it may be okay to keep it simple with just a single container app. Thoughts?